### PR TITLE
Pinning tweaks

### PR DESCRIPTION
**Pull request by jakirkham**
Resolve some errata that was uncovered in the pinnings from PR ( https://github.com/conda-forge/conda-forge.github.io/pull/156 ).

### CODE REVIEW

File: scripts/pin_the_slow_way.py [hunk 1]
```diff
@@ -33,15 +33,15 @@
...
           'freetype': 'freetype 2.6.*',
-          'hdf5': 'hdf5 1.8.17',
+          'hdf5': 'hdf5 1.8.17|1.8.17.*',
           'icu': 'icu 56.*',
           'jpeg': 'jpeg 9*',
           'libnetcdf': 'libnetcdf 4.4.*',
           'libpng': 'libpng >=1.6.21,<1.7',
           'libtiff': 'libtiff 4.0.*',
-          'ncurses': 'ncurses 5.9.*',
+          'ncurses': 'ncurses 5.9*',
```
Comment (pelson) [inline scripts/pin_the_slow_way.py]: 5.90 a risk :smile:?

File: scripts/pin_the_slow_way.py [hunk 2]
```diff
@@ -33,15 +33,15 @@
...
           'openssl': 'openssl 1.0.*',
-          'readline': 'readline 6.2.*',
+          'readline': 'readline 6.2*',
```
Comment (pelson) [inline scripts/pin_the_slow_way.py]: 6.20 a risk?

File: scripts/pin_the_slow_way.py (hunk 1 above)
Comment (janschulz) [inline scripts/pin_the_slow_way.py]: Just a question: what's actually so bad about using the dot? Or does `5.9*` matches 5.10? Or is `.` interpreted as `anything` like in a regex?

Comment (pelson) [inline scripts/pin_the_slow_way.py]: The problem with this case is that there is no `5.9.0` it only has two numbers `5.9`. Taking this to its conclusion, it makes sense to simply say `5.9` if that is what we mean.

**Pull request merged**